tests/integration/test_exoduslambda.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+import logging
+import requests
+
+from .utils import get_config, get_db_client
+from boto3.dynamodb.conditions import Key, Attr
+
+
+class TestExodus:
+    def setup(self):
+        self.config = get_config()
+        self.db_client = get_db_client()
```

Comment:
Can we please rewrite this in a way which doesn't require access to the DB?  I would like these tests to be usable from the outside without any special permissions, as long as you have the ability to fetch from the CDN.

Using the DB to find URLs for test in this way is also kinda risky since you could potentially get a different URL each time, leading to unstable test results.  And expensive, at least the way you're doing it now (table scan means reading every row in the table, and this table will be huge on production).

My suggestion is, instead of using the DB to find URLs for test, please simply hardcode the paths for test.  We can call those paths the "reference testdata" and require that every environment has that data in order to support these tests.

Reply:
I believe the "reference testdata" can make same test suite capable for different environments and help manage the test_data, but it also limit the test coverage. I am not sure about it. Is using DB really that unacceptable?

Reply:
Yeah, you really shouldn't use the DB to drive these tests.

I don't see how it limits the test coverage to avoid using the DB - you don't even know which URLs are tested with the current approach, do you?  So you don't actually know what's covered right now...

As an example of the flaws with the DB-driven approach: the lambda only responds with data having a from_date in the past.  Your test code querying the DB didn't take this into account, so the test could find URLs which are not yet exposed and then the test would incorrectly fail.  To "fix" that with current approach, you'd essentially need the test to reproduce the same logic as used within the lambda and then maintain the two of them together.  I don't think that makes sense... it is too risky that the same bug is written into both the test and the code under test.